src/hotspot/share/opto/parse1.cpp

```diff
@@ -1211,6 +1212,10 @@ void Parse::do_method_entry() {
     make_dtrace_method_entry(method());
   }
 
+  if (CDSConfig::is_dumping_preimage_static_archive_with_triggers() && method()->is_end_training_trigger()) {
```

Comment:
Do we come here also if `method()` is being inlined by C2?

Reply:
It does, I tested inlining manually and proved that the method was inlined and still triggering, I'll see if I can add a test case so that if the inlining code path changes the test failure will direct us to add the appropriate code to the new code path